core/java/jdwp/cmdsets.go

```diff
@@ -36,3 +50,249 @@ const (
+	cmdMethodTypeBytecodes                = cmd{cmdSetMethod, 3}
+	cmdMethodTypeIsObsolete               = cmd{cmdSetMethod, 4}
+	cmdMethodTypeVariableTableWithGeneric = cmd{cmdSetMethod, 5}
+
+	cmdObjectReferenceReferenceType     = cmd{cmdSetObjectReference, 1}
+	cmdObjectReferenceGetValues         = cmd{cmdSetObjectReference, 2}
+	cmdObjectReferenceSetValues         = cmd{cmdSetObjectReference, 3}
+	cmdObjectReferenceMonitorInfo       = cmd{cmdSetObjectReference, 5}
+	cmdObjectReferenceInvokeMethod      = cmd{cmdSetObjectReference, 6}
+	cmdObjectReferenceDisableCollection = cmd{cmdSetObjectReference, 7}
+	cmdObjectReferenceEnableCollection  = cmd{cmdSetObjectReference, 8}
+	cmdObjectReferenceIsCollected       = cmd{cmdSetObjectReference, 9}
+
+	cmdStringReferenceValue = cmd{cmdSetStringReference, 1}
+
+	cmdThreadReferenceName                    = cmd{cmdSetThreadReference, 1}
+	cmdThreadReferenceSuspend                 = cmd{cmdSetThreadReference, 2}
+	cmdThreadReferenceResume                  = cmd{cmdSetThreadReference, 3}
+	cmdThreadReferenceStatus                  = cmd{cmdSetThreadReference, 4}
+	cmdThreadReferenceThreadGroup             = cmd{cmdSetThreadReference, 5}
+	cmdThreadReferenceFrames                  = cmd{cmdSetThreadReference, 6}
+	cmdThreadReferenceFrameCount              = cmd{cmdSetThreadReference, 7}
+	cmdThreadReferenceOwnedMonitors           = cmd{cmdSetThreadReference, 8}
+	cmdThreadReferenceCurrentContendedMonitor = cmd{cmdSetThreadReference, 9}
+	cmdThreadReferenceStop                    = cmd{cmdSetThreadReference, 10}
+	cmdThreadReferenceInterrupt               = cmd{cmdSetThreadReference, 11}
+	cmdThreadReferenceSuspendCount            = cmd{cmdSetThreadReference, 12}
+
+	cmdThreadGroupReferenceName     = cmd{cmdSetThreadGroupReference, 1}
+	cmdThreadGroupReferenceParent   = cmd{cmdSetThreadGroupReference, 2}
+	cmdThreadGroupReferenceChildren = cmd{cmdSetThreadGroupReference, 3}
+
+	cmdArrayReferenceLength    = cmd{cmdSetArrayReference, 1}
+	cmdArrayReferenceGetValues = cmd{cmdSetArrayReference, 2}
+	cmdArrayReferenceSetValues = cmd{cmdSetArrayReference, 3}
+
+	cmdClassLoaderReferenceVisibleClasses = cmd{cmdSetClassLoaderReference, 1}
+
+	cmdEventRequestSet                 = cmd{cmdSetEventRequest, 1}
+	cmdEventRequestClear               = cmd{cmdSetEventRequest, 2}
+	cmdEventRequestClearAllBreakpoints = cmd{cmdSetEventRequest, 3}
+
+	cmdStackFrameGetValues  = cmd{cmdSetStackFrame, 1}
+	cmdStackFrameSetValues  = cmd{cmdSetStackFrame, 2}
+	cmdStackFrameThisObject = cmd{cmdSetStackFrame, 3}
+	cmdStackFramePopFrames  = cmd{cmdSetStackFrame, 4}
+
+	cmdClassObjectReferenceReflectedType = cmd{cmdSetClassObjectReference, 1}
+
+	cmdEventComposite = cmd{cmdSetEvent, 1}
+)
+
+var cmdNames = map[cmd]string{}
+
+func init() {
+	register := func(c cmd, n string) {
+		if _, e := cmdNames[c]; e {
+			panic("command already registered")
+		}
+		cmdNames[c] = n
+	}
+	register(cmdVirtualMachineVersion, "Version")
```

Comment:
why not just make the name a field in the cmd struct?

Reply:
Because if we get a cmdSet and cmdID from somewhere else (say the wire), we can use struct-equality to lookup the name.

Reply:
Thanks